src/implementation/Server/HTTPServer/pubsub.ts

```diff
@@ -27,13 +28,13 @@ export default class HTTPServerPubSub implements IServerPubSub {
     this.logger = new Logger("HTTPServer", "PubSub", server.client.options.logger);
   }
 
-  async subscribe(pubsubName: string, topic: string, cb: TypeDaprPubSubCallback, route = "") {
+  async subscribe(pubsubName: string, topic: string, cb: TypeDaprPubSubCallback, route = "", metadata?: KeyValueType) {
     if (!route) {
       route = `route-${pubsubName}-${topic}`;
     }
 
     // Register the handler
-    await this.server.getServerImpl().registerPubSubSubscriptionRoute(pubsubName, topic, route);
+    await this.server.getServerImpl().registerPubSubSubscriptionRoute(pubsubName, topic, route, metadata ?? {});
```

Comment:
Shouldn't we just pass metadata and make it default in the `registerPubSubSubscriptionRoute`?